src/main/kotlin/org/web3j/evm/InMemoryBesuChain.kt

```diff
@@ -165,14 +166,14 @@ class InMemoryBesuChain(
      * @return the transaction receipt
      */
     fun processTransaction(transaction: Transaction): TransactionReceipt {
-        val nextBlockNumber = blockChain.chainHeadBlockNumber + 1
-        val spec = protocolSchedule.getByBlockNumber(nextBlockNumber)
+        val headHeader = blockChain.chainHeadHeader
```

Comment:
It's been a while since I've looked at this, so just checking the reason for the change between +1 and not doing +1 in the change?